crates/modelardb_server/src/storage/compressed_data_manager.rs

```diff
@@ -328,17 +343,20 @@ impl CompressedDataManager {
 
         let file_path = compressed_data_buffer.save_to_apache_parquet(folder_path.as_path())?;
 
-        // Update the remaining memory for compressed data and record the change.
+        // unwrap() is safe as lock() only returns an error if the lock is poisoned.
         let freed_memory = compressed_data_buffer.size_in_bytes;
-        self.memory_pool.free_compressed_memory(freed_memory);
         self.used_compressed_memory_metric
+            .lock()
+            .unwrap()
             .append(-(freed_memory as isize), true);
 
         // Record the change to the used disk space.
         let file_size = file_path.metadata()?.len() as isize;
+
+        // unwrap() is safe as lock() only returns an error if the lock is poisoned.
         self.used_disk_space_metric
-            .write()
-            .await
+            .lock()
+            .unwrap()
             .append(file_size, true);
 
         debug!(
```

Comment:
Should this debug message be moved down so it comes after we free the compressed memory?

Reply:
Fixed.